### ISSUE TITLE
Workspaces no longer correctly identified since 5.10

### ISSUE THREAD

**Issue by bbeesley** [webpro/knip#617]
Updating to versions greater than 5.9.4 gives us thousands of errors. Running with the debug flag shows the reason for this, our workspaces are no longer properly identified. In the debug output I see the following:

```
[*] Included workspaces
[
  'root',
  'NOT_FOUND_packages/protobuf',
  'NOT_FOUND_packages/xoconfig',
  'NOT_FOUND_packages/tsconfig',
  'NOT_FOUND_packages/srv-http',
  'NOT_FOUND_packages/srv-utils',
  'NOT_FOUND_packages/srv-fetch',
  'NOT_FOUND_packages/srv-logger',
  'NOT_FOUND_packages/vas-clients',
  'NOT_FOUND_packages/srv-graphql',
  'NOT_FOUND_packages/fastify-utils',
  'NOT_FOUND_packages/srv-base-conf',
  'NOT_FOUND_packages/srv-idl-client',
  'NOT_FOUND_packages/srv-admin-server',
  'NOT_FOUND_packages/srv-backstage-utils',
  '@our-npm-org/xoconfig',
  '@our-npm-org/tsconfig',
  '@our-npm-org/srv-http',
  '@our-npm-org/protobuf',
  '@our-npm-org/srv-utils',
  '@our-npm-org/srv-fetch',
  '@our-npm-org/srv-logger',
  '@our-npm-org/vas-clients',
  '@our-npm-org/srv-graphql',
  '@our-npm-org/srv-base-conf',
  '@our-npm-org/fastify-utils',
  '@our-npm-org/srv-idl-client',
  '@our-npm-org/srv-admin-server',
  '@our-npm-org/srv-backstage-utils'
]
```

Workspace config in our root package json looks like this:

```
  "workspaces": [
    "./packages/*"
  ],
```

The name field in each of the workspace's package.json files includes the namespace for our npm org, which I assume is where its finding those names. But they don't map to what we have defined in config, which looks like:

```
  "workspaces": {
    "packages/*": {
      "entry": ["src/index.ts"],
      "project": ["src/*.ts", "src/**/*.ts"],
    },
    "packages/srv-graphql": {
      "entry": ["src/index.ts"],
      "project": ["src/*.ts", "src/**/*.ts"],
      "ignoreDependencies": ["prettier-2"]
    },
```

Holler if you need anymore information.

**Comment by webpro** [comment]
Thanks for the report. Does it work like before if you use `"workspaces": ["packages/*"]` in `package.json` - without the `./` prefix?

**Comment by glemiron** [comment]
I don't sure if it related, but I've mentioned today that the config ignored if I run default `npx knip`. In the same time `knip --include files` works correct and uses the config. 

It was easy to spot, because I use jest plugin for visual tests, but I have visual tests as unused files when I run default option. 

`npx knip --include files,exports,types` also doesn't work as expected

**Comment by bbeesley** [comment]
> Thanks for the report. Does it work like before if you use `"workspaces": ["packages/*"]` in `package.json` - without the `./` prefix?

@webpro Yeah it does actually. That doesn't appear to affect anything else obvious so probably we can do that for now. I'd still suggest this needs a fix though since the leading `./` is [what they show in the npm docs](https://docs.npmjs.com/cli/v10/configuring-npm/package-json#workspaces).

**Comment by webpro** [comment]
Totally agree, just wanted to make sure what is exactly the issue here.

**Comment by webpro** [comment]
:rocket: _This issue has been resolved in v5.12.1. See [Release 5.12.1](https://github.com/webpro/knip/releases/tag/5.12.1) for release notes._

_Using Knip in a commercial project? Please consider [sponsoring me](https://github.com/sponsors/webpro)._
